### ISSUE TITLE
Ensure ECMAScript track is ready for v2 launch

### ISSUE THREAD

**Issue by kytrinyx** [exercism/javascript#412]
There are a number of things we're going to want to check before the v2 site goes live. There are notes below that flesh out all the checklist items.

- [x] The track has a page on the v2 site: https://v2.exercism.io/tracks/ecmascript
- [x] The track page has a short description under the name (not starting with `TODO`)
- [x] The "About" section is a friendly, colloquial, compelling introduction
- [x] The "About" section follows the formatting guidelines
- [x] The code example gives a good taste of the language and fits within the boundaries of the background image
- [x] There are exercises marked as `core`
- [x] Exercises have rough estimates of difficulty
- [x] Exercises have topics associated with them
- [x] The first exercise is `auto_approve: true`

## Track landing page

The v2 site has a landing page for each track, which should make people want to join it. If the track page is missing, ping `@kytrinyx` to get it added.

### Blurb
If the header of the page starts with `TODO`, then submit a pull request to https://github.com/exercism/ecmascript/blob/master/config.json with a `blurb` key.  Remember to get [configlet](https://github.com/exercism/configlet/releases/latest) and run `configlet fmt .` from the root of the track before submitting.

### About section
If the "About" section feels a bit dry, then submit a pull request to https://github.com/exercism/ecmascript/blob/master/docs/ABOUT.md with suggested tweaks.

#### Formatting guidelines
In order to work well with the design of the new site, we're restricting the formatting of the `ABOUT.md`. It can use:

* Bold
* Italics
* Links
* Bullet lists
* Number lists

Additionally:

* Each sentence should be on its own line
* Paragraphs should be separated by an empty line
* Explicit `<br/>` can be used to split a paragraph into lines without spacing between them, however this is discouraged.

### Code example
If the code example is too short or too wide or too long or too uninteresting, submit a pull request to https://github.com/exercism/ocaml/blob/master/docs/SNIPPET.txt with a suggested replacement.

## Exercise metadata
Where the v1 site has a long, linear list of exercises, the v2 site has organized exercises into a small set of required exercises ("core").

If you update the track config, remember to get [configlet](https://github.com/exercism/configlet/releases/latest) and run `configlet fmt .` from the root of the track before submitting.

### Topic and difficulty
Core exercises unlock optional additional exercises, which can be filtered by topic an difficulty, however that will only work if we add topics and difficulties to the exercises in the track config, which is in https://github.com/exercism/ecmascript/blob/master/config.json

### Auto-approval
We've currently made any hello-world exercises auto-approved in the backend of v2. This means that you don't need mentor approval in order to move forward when you've completed that exercise.

Not all tracks have a hello-world, and some tracks might want to auto approve other (or additional) exercises.

## Track mentors
There are no bullet points for this one :)

As we move towards the launch of the new version of Exercism we are going to be ramping up on actively recruiting people to help provide feedback. Our goal is to get to 100%: everyone who submits a solution and wants feedback should get feedback. Good feedback.

If you're interested in helping mentor the track, check out http://mentoring.exercism.io/

#
When all of the boxes are ticked off, please close the issue.

Tracking progress in https://github.com/exercism/meta/issues/104

**Comment by jackhughesweb** [comment]
Things that can be already checked off:
> The track has a page on the v2 site: https://v2.exercism.io/tracks/ecmascript
> There are exercises marked as core
> Exercises have rough estimates of difficulty
> Exercises have topics associated with them

I've submitted pull requests for the blurb #413, and the about section #414.

The first exercise is missing "auto_approve", but has "hello-world" as the name of the first exercise - this may need to be fixed later.

**Comment by matthewmorgan** [comment]
@jackhughesweb thanks for your work on this.  It does raise the question of whether it is time to deprecate this track, and move the ES features in to the JavaScript track.

Arguments in favor:
1. ES6 is now 3 years old.
2. Transpilation is nearly a ubiquitous part of the modern JavaScript development workflow.

@exercism/ecmascripts would anyone care to comment on this, pro or con?

**Comment by tejasbubane** [comment]
I agree with @matthewmorgan. Ecmascript has grown a lot over the last few years almost all developers are writing ecmascript these days. Makes very little sense now to maintain two different tracks.

And the release of exercism v2 is a nice opportunity to do the transition 😃

**Comment by matthewmorgan** [comment]
@tejasbubane my thoughts exactly. :)

**Comment by joelwallis** [comment]
Same here. No reason to keep two tracks, and Nextercism is the perfect
opportunity to unite them.

Should we plan an effort to merge the tracks? Maybe we could add some
improvements as general acceptance criteria.

Joel Wallis Jucá
joelwallis.com

Em ter, 12 de jun de 2018 2:30 PM, Matthew Morgan <notifications@github.com>
escreveu:

**Comment by kytrinyx** [comment]
Excellent! I'll take a bit of time to think through the details and chat with @iHiD and @nicolechalmers about how to proceed, and get back to you with some concrete suggestions.

Meanwhile, if you have thoughts/ideas about things we should remember to do, jot them down here!

**Comment by matthewmorgan** [comment]
Copying here my recent email to @kytrinyx about this, looking for others' comments and help:

> Off the top of my head, one possible approach would be to replace the JS track with the ES one, adding back in any exercises implemented in JS but not in ES.  This would surely require some users to rewrite their solutions, but this would be no different than when a test suite gets changed now.  So, perhaps this is step 1, and maybe step 0 is an audit to determine what stragglers need to be ported.
  Step 2 could be updating the JS-flavored examples to ES.
  Somewhere in there we'll need to update the docs to say 'JavaScript' everywhere, and maybe add a note about the ES track having been folded in to JS.
  

> I don't know what this means for the UI app though-- do we need to preserve the UUIDs from the JS exercises, or is that just for V2?  Will the slugs being the same be enough to map users' existing work and comments to the updated exercises?

**Comment by matthewmorgan** [comment]
After more conversation with @kytrinyx and others, I've prepared a few things for deprecating the current JS track.

For V2, the EcmaScript track will become the "new" JavaScript track.  Users' submissions will be migrated as part of the process.

Katrina's notes on the migration process:

> We will take the v1 site down in order to launch v2.
> The site will have to be offline for 24-48 hours, because we have a massive data migration that we need to do.
> 
> During that time, we can:
> 
> - rename https://github.com/exercism/javascript to https://github.com/exercism/DEPRECATED.javascript
> - set `active: false` in the config of DEPRECATED.javascript
> - rename https://github.com/exercism/ecmascript to https://github.com/exercism/javascript
> - delete the GitHub team exercism/javascript
> - rename the GitHub team exercism/ecmascript to exercism/javascript

In the meantime, I've closed all open issues and PRs on the JS track as `wontfix`, leaving one issue open stating the track will be deprecated.  I have also updated the README to alert users to the deprecation.

On this track, we need to 
[ ] identify any exercises that exist in the current JS track but _not_ in this one
[ ] open issues to port those exercises

@tejasbubane @joelwallis ( @jackhughesweb ? ) would any of you care to take on the audit of missing exercises?  That step seems to need to be done first.

**Comment by joelwallis** [comment]
Matt,

I could help on it. Should I compare the implementations too? EcmaScropt
and JavaScript tracks have different ways to implement the exercises. The
first one uses way more classes while the second one uses functions and
constructors. How will we proceed with it? Is it something relevant?

Joel Wallis Jucá
joelwallis.com

Em sáb, 23 de jun de 2018 1:15 PM, Matthew Morgan <notifications@github.com>
escreveu:

**Comment by matthewmorgan** [comment]
@joelwallis I'll respond on #417

**Comment by iHiD** [comment]
Great work. Thank you.

I think @kytrinyx and I need an hour to plan, then probably an three hours to code a script, then deploy and check. We need some info on people mid-way through an exercise too and to think about how the CLI copes with that.

We have to launch Teams before then so I think we're probably targeting maybe around 1st-3rd Aug to do it. If things go badly then maybe towards the next week, although I'll be flat out with the day-job then so I'd like to get it done by the 3rd.

**Comment by kytrinyx** [comment]
Yeah 1st-3rd of August would be great. Let me find those formatting guidelines (and let me stick those in some proper documentation!)

**Comment by kytrinyx** [comment]
These are the guidelines for the ABOUT.md file: https://github.com/exercism/docs/blob/master/language-tracks/documentation/for-consumers.md#about

**Comment by matthewmorgan** [comment]
OK thanks.  `ABOUT.md` conforms to the requirements.

**Comment by tejasbubane** [comment]
This track along with merger of ecmascript has been running well on v2. Closing. 

Please reopen if something is missed.
